src/Application.vala

```diff
@@ -108,9 +108,11 @@ public class AppCenter.App : Granite.Application {
         }
 
         if (main_window == null) {
-            client.update_cache.begin (true);
-
             main_window = new MainWindow (this);
+            // Start updating the cache 5 seconds after starting the main window
+            // If we do it instantly, we slow down the loading of the banner on the
+            // homepage considerably
+            schedule_cache_update (false, 5);
```

Comment:
Not really, it just prevents the banner thread and the main thread interfering when talking to PackageKit. I guess there's a couple of options:

One is to move that code out to Homepage.vala and trigger it after the banner has finished loading, but I guess you'd expect that code to be in Application.vala instead of hidden in one of the Widgets. The other is to just make the number less "magic" and give it a const.